consume_test.go

```diff
@@ -797,23 +816,78 @@ func TestConsumeParseArgsDefault(t *testing.T) {
 	defer c.Done()
 
 	c.Setenv("KT_BROKERS", "")
+	c.Setenv("KT_REGISTRY", "")
+
 	cmd0, _, err := parseCmd("hkt", "consume")
 	c.Assert(err, qt.Equals, nil)
 	cmd := cmd0.(*consumeCmd)
 	c.Assert(cmd.brokers(), qt.DeepEquals, []string{"localhost:9092"})
+	c.Assert(cmd.registryURL, qt.Equals, "")
 }
 
 func TestConsumeParseArgsFlagsOverrideEnv(t *testing.T) {
 	c := qt.New(t)
 	defer c.Done()
 
+	registry := "localhost:8084"
+	broker := "hans:2000"
+
 	// command line arg wins
 	c.Setenv("KT_BROKERS", "BLABB")
+	c.Setenv("KT_REGISTRY", "BLABB")
 
-	cmd0, _, err := parseCmd("hkt", "consume", "-brokers", "hans:2000")
+	cmd0, _, err := parseCmd("hkt", "consume", "-brokers", broker, "-registry", registry)
 	c.Assert(err, qt.Equals, nil)
 	cmd := cmd0.(*consumeCmd)
-	c.Assert(cmd.brokers(), qt.DeepEquals, []string{"hans:2000"})
+	c.Assert(cmd.brokers(), qt.DeepEquals, []string{broker})
+	c.Assert(cmd.registryURL, qt.Equals, registry)
+}
+
+func TestConsumeAvroMessage(t *testing.T) {
+	c := qt.New(t)
+	defer c.Done()
+
+	type record struct {
+		A int
+		B int
+	}
+
+	// In the byte slice below:
+	//	80: A=40
+	//	40: B=20
+	rec := record{A: 40, B: 20}
+	data := []byte{80, 40}
+
+	_, wType, err := avro.Marshal(rec)
+	c.Assert(err, qt.IsNil)
+
+	reg := newTestRegistry(c)
+	schemaID := reg.register(c, wType)
+
+	cmd := consumeCmd{registry: reg.registry}
+
+	enc, err := cmd.encoderForType("string")
+	c.Assert(err, qt.IsNil)
+	cmd.encodeKey = enc
+
+	enc, err = cmd.encoderForType("avro")
+	c.Assert(err, qt.IsNil)
+	cmd.encodeValue = enc
+
+	msg := &sarama.ConsumerMessage{
+		Key:       []byte("foo"),
+		Value:     createAvroMessage(schemaID, data),
+		Partition: 1,
+		Offset:    0,
+	}
+
+	consumed, err := cmd.newConsumedMessage(msg)
+	c.Assert(err, qt.IsNil)
+
+	expected := record{}
+	err = json.Unmarshal(consumed.Value, &expected)
+	c.Assert(err, qt.IsNil)
+	c.Assert(rec, qt.DeepEquals, expected)
```

Comment:
Isn't this the wrong way round? i.e. `rec` holds the expected value and `expected` holds the value that we're checking.

I think I might do:

```
var got record
err = json.Unmarshal(consumed.Value, &got)
c.Assert(err, qt.IsNil)
c.Assert(got, qt.DeepEquals, rec)
```